### ISSUE TITLE
cobbler sync not prefixing https when http_port is 443

### ISSUE THREAD

**Issue by dansou901** [cobbler/cobbler#2418]
### Describe the bug

When using https for cobbler and having http_port set to 443, the url prefix in the urls present in kernel options in pxelinux.cfg/default stays http, but should be set to https. This results in errors during autoinstallation file fetching.

### To Reproduce

1. Use https for cobbler web interface and api
2. Set http_port to 443 in settings
3. Define profiles with autoinstallation files
4. Run cobbler sync
5. Check contents of pxelinux.cfg/default

### Expected behavior

The protocol prefix in the URL should be https when http_port is set to 443.

### Actual behavior

The protocol prefix in the URL stays at http.

### Version

Cobbler version is 3.1.2 in my environment.

**Comment by SchoolGuy** [comment]
So we have various places which contain hardcoded `http` URLs. Thus I think this is not limited to kernel parameters. Deeper investigation is needed since the template itself is not containing the lock on the `http`.

**Comment by dansou901** [comment]
There is an option called "client_use_https". When that is set to 1, all cobbler urls which are getting called anywhere should be changed to https.

**Comment by SchoolGuy** [comment]
@dansou901 I agree that this should be the case but apparently we did not implement that everywhere. If I fixed some other (in my opinion) more important tech-debt (PR will come through later today), then I take care of this. Sorry to not be able to help faster.

**Comment by dansou901** [comment]
@SchoolGuy don't worry too much about this one, if there are other ones to fix which are more urgent, go for it. I've got a workaround so I'm fine for the time being, this is just for completeness.
